pkg/vm/vm_test.go

```diff
@@ -346,34 +348,168 @@ func TestInferOperandSub(t *testing.T) {
 
-	cellOp0 := &mem.Cell{
-		Accessed: true,
-		Value: mem.MemoryValueFromMemoryAddress(
-			mem.NewMemoryAddress(2, 10),
-		),
-	}
+	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.NoError(t, err)
+	assert.Nil(t, res)
+}
 
-	cellOp1 := &mem.Cell{
-		Accessed: true,
-		Value:    mem.MemoryValueFromInt(15),
-	}
+func TestComputeRes_Op1(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: Op1}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(10)}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
 
 	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
 	require.NoError(t, err)
+	expected := mem.MemoryValueFromInt(15)
+	assert.Equal(t, expected, res)
+}
 
-	expected := mem.MemoryValueFromMemoryAddress(
-		mem.NewMemoryAddress(2, 25),
-	)
+func TestComputeAddRes_AddrToFelt(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: AddOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10))}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
+
+	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.NoError(t, err)
+	expected := mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 25))
+	assert.Equal(t, expected, res)
+}
+
+func TestComputeAddRes_FeltToAddr(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: AddOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 15))}
+
+	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.NoError(t, err)
+	expected := mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 30))
+	assert.Equal(t, expected, res)
+}
+
+func TestComputeAddRes_BothAddrs(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: AddOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10))}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 15))}
+
+	_, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.Error(t, err) // Expecting an error since adding two addresses is not allowed
+}
+
+func TestComputeAddRes_BothFelts(t *testing.T) {
+	vm := defaultVirtualMachine()
+	instruction := Instruction{Res: AddOperands}
+
+	cellOp0 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(10)}
+	cellOp1 := &mem.Cell{Accessed: true, Value: mem.MemoryValueFromInt(15)}
+
+	res, err := vm.computeRes(&instruction, cellOp0, cellOp1)
+	require.NoError(t, err)
+	expected := mem.MemoryValueFromInt(25)
+	assert.Equal(t, expected, res)
+}
+
+// Felt should be Positive or Negative. Thus four test cases
+func TestComputeMulRes_PToPFelt(t *testing.T) {
```

Comment:
```suggestion
func TestComputeMulResPosToPosFelt(t *testing.T) {
```

Reply:
Done. Applied to other function names too.